### PR TITLE
Adding the add and remove actionlets

### PR DESCRIPTION
**Pull request by jdotcms**

### CODE REVIEW

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java [hunk 1]
```diff
@@ -146,6 +162,66 @@ public final Response findSchemes(@Context final HttpServletRequest request,
...
+     * @return Response
+     */
+    @GET
+    @Path("/actionlets")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findActionlets(@Context final HttpServletRequest request) {
+
+        final InitDataObject initDataObject = this.webResource.init
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java]: done

File: dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionBean.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            this.actionId = actionId;
+            return this;
+        }
+
+        public Builder actionletClass(String actionletClass) {
+            this.actionletClass = actionletClass;
+            return this;
+        }
+
+        public Builder order(int order) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionBean.java]: done

File: dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionBean.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            this.order = order;
+            return this;
+        }
+
+        public Builder parameters(Map<String, String> parameters) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionBean.java]: done

File: dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionForm.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        this.actionletClass = builder.actionletClass;
+        this.order          = builder.order;
+        this.parameters     = builder.parameters;
+        this.checkValid();
+    }
+
+    public static final class Builder {
+
+        @JsonProperty()
+        private int order = 0;
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionForm.java]: Yeah is a builder

File: dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionForm.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        @JsonProperty(required = true)
+        private String        actionletClass;
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/workflow/form/WorkflowActionletActionForm.java]: it is a builder so can not do it

File: dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java [hunk 6]
```diff
@@ -1381,10 +1353,90 @@ public WorkflowAction saveAction(final WorkflowActionForm workflowActionForm, fi
     }
 
     @WrapInTransaction
-    public void saveActionToStep(final WorkflowActionStepBean workflowActionStepForm, final User user) throws DotDataException, DotSecurityException {
+    public void saveActionletToAction(final WorkflowActionletActionBean workflowActionletActionBean, final User user)
+            throws DotDataException, AlreadyExistException {
+
+        final String actionId       = workflowActionletActionBean.getActionId();
+        final String actionletClass = workflowActionletActionBean.getActionletClass();
+        final int    order          = workflowActionletActionBean.getOrder();
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java]: Done

**Pull request merged**